libcudacxx/include/cuda/std/__cstdlib/abs.h

```diff
@@ -0,0 +1,43 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___CSTDLIB_ABS_H
+#define _LIBCUDACXX___CSTDLIB_ABS_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI constexpr int abs(int __val) noexcept
```

Comment:
Just a comment that we are all on the same page here.

There are compiler builtins for this that are always optimized even in debug mode. However, those are not constexpr and this is a function we kind of want to be constexpr.

So in this case the implementation is so simple we decided to implement on our own

Reply:
The alternative was something like:
```cpp
_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_INT_ABS int abs(int __val) noexcept
{
#if !_CCCL_HAS_CONSTEXPR_BUILTIN_ABS
  if (_CUDA_VSTD::is_constant_evaluated())
  {
    return (__val < 0) ? -__val : __val;
  }
#endif // !_CCCL_HAS_CONSTEXPR_BUILTIN_ABS

#if defined(_CCCL_BUILTIN_ABS)
  return _CCCL_BUILTIN_ABS(__val);
#else // ^^^ _CCCL_BUILTIN_ABS ^^^ / vvv !_CCCL_BUILTIN_ABS vvv
  return ::abs(__val);
#endif // !_CCCL_BUILTIN_ABS
}
```

Reply:
I think we can skip calling the built-in here, because it would make the implementation more complex for arguably no gain. AFAIK, the purpose of the builtin is to provide optimized codegen also in debug builds. But I doubt that perf for `(__val < 0) ? -__val : __val` will be terrible for built-in types in debug builds.